elastic/quark
# Introduce build target initramfs.gz

`make initramfs.gz` builds an `initramfs.gz` that contains all quark utilities statically compiled, as well as a super simple init that sets the environment so that quark can run.

The purpose is to be able to test different kernels and run `quark-mon` or `quark-btf` as you run normally. This is the main infrastructure for the future `quark-test` which will be a single binary that tests "everything", it forks, execs, and whatnot.

How to test it:
$ make initramfs.gz
$ qemu-system-x86_64 -initrd initramfs.gz -kernel linux-image-x86_64-5.10.92-2 \
	-nographic --append "console=ttyS0 quark-mon -kvvv"

You can pass parameters normally to the binary, see `append` above.
